matching-notification-api/src/main/java/org/phenotips/matchingnotification/storage/internal/DefaultMatchStorageManager.java

```diff
@@ -285,23 +285,16 @@ public boolean saveRemoteMatches(List<? extends PatientSimilarityView> similarit
             query.setParameter("phenScore", phenScore);
             query.setParameter("genScore", genScore);
 
-            SimpleDateFormat sdf = new SimpleDateFormat("yyyy/MM/dd");
-            if (StringUtils.isNotBlank(fromDate)) {
-                try {
-                    Timestamp timestampFrom = new Timestamp(sdf.parse(fromDate).getTime());
-                    query.setTimestamp("fromTimestamp", timestampFrom);
-                } catch (Exception e) {
-                    // do nothing
-                }
+            if (fromDate != null) {
+                Timestamp timestampFrom = new Timestamp(fromDate.getTime());
+                query.setTimestamp("fromTimestamp", timestampFrom);
+
             }
 
-            if (StringUtils.isNotBlank(toDate)) {
-                try {
-                    Timestamp timestampTo = new Timestamp(sdf.parse(toDate).getTime());
-                    query.setTimestamp("toTimestamp", timestampTo);
-                } catch (Exception e) {
-                    // do nothing
-                }
+            if (toDate != null) {
+                // need to add 1 day in ms to include all matches made that day
+                Timestamp timestampTo = new Timestamp(toDate.getTime() + 86400000);
```

Comment:
It is better to write `1000*60*60*24` than `86400000`, then it is clear where the number comes from